### PR TITLE
Getting started guide, Java version

### LINKED ISSUE #354: Java sample for GettingStartedGuide
The Getting Started Guide needs java code samples.

### PR DESCRIPTION
**Pull request by seglo**
Fixes #354

### CODE REVIEW

File: examples/src/test/java/jdocs/guide/EventGeneratorApp.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      EntityTypeKey.create(ShoppingCartEvents.Event.class, "shopping-cart-event");
+
+  static Behavior<Command> create(Boolean clusterMode) {
+
+    return Behaviors.setup(
+        context -> {
+          ActorSystem<Void> system = context.getSystem();
+          Cluster cluster = Cluster.get(system);
+
+          ActorRef<SelfUp> upAdapter = context.messageAdapter(SelfUp.class, up -> new Start());
```
Comment (patriknw) [inline examples/src/test/java/jdocs/guide/EventGeneratorApp.java:81]: It's not necessary to wait until Up. Can be simplified by removing this and the Start message.

File: examples/src/test/java/jdocs/guide/ShoppingCartTags.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+// #guideTags
+package jdocs.guide;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ShoppingCartTags {
+  public static String Single = "shopping-cart";
+  public static List<String> Tags = Arrays.asList("carts-0", "carts-1", "carts-2");
```
Comment (patriknw) [inline examples/src/test/java/jdocs/guide/ShoppingCartTags.java:13]: maybe use Java convention for constants: SINGLE, TAGS

File: examples/src/test/java/jdocs/guide/EventGeneratorApp.java (hunk 1 above)
Comment (seglo) [reply examples/src/test/java/jdocs/guide/EventGeneratorApp.java]: :+1:

File: examples/src/test/java/jdocs/guide/ShoppingCartTags.java (hunk 2 above)
Comment (seglo) [reply examples/src/test/java/jdocs/guide/ShoppingCartTags.java]: :+1:

**Comment by seglo** [comment]
TODO: fix build failures for Java 8 compatibility.

File: examples/src/test/java/jdocs/guide/ShoppingCartEvents.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    public CheckedOut(String cartId, Instant eventTime) {
+      this.cartId = cartId;
+      this.eventTime = eventTime;
+    }
+
+    public String getCartId() {
+      return this.cartId;
+    }
+  }
+}
```
Comment (patriknw) [inline examples/src/test/java/jdocs/guide/ShoppingCartEvents.java:94]: @seglo which events couldn't be serialized? Two typical gotchas:
* constructor with single parameter: https://doc.akka.io/docs/akka/current/serialization-jackson.html#constructor-with-single-parameter
* `-parameters` Java compiler option https://github.com/akka/akka-samples/blob/2.6/akka-sample-cqrs-java/pom.xml#L136 or for sbt `.settings(javacOptions += "-parameters") // for Jackson`

Comment (seglo) [reply examples/src/test/java/jdocs/guide/ShoppingCartEvents.java:94]: It was a deserializable problem with the Java `ShoppingCartApp`.  The `javacOptions += "-parameters"` setting resolved it.  Thank you!

**Comment by seglo** [comment]
Ready for review.

File: docs/src/main/paradox/getting-started/running.md [hunk 4]
```diff
@@ -60,31 +60,29 @@ Add the `EventGeneratorApp` to your project:
 Scala
```
Comment (patriknw) [inline docs/src/main/paradox/getting-started/running.md:60]: Somewhere in this page it says "Akka Cluster Sharding Typed". Normally we don't include Typed in such module names. Great if you can change to "Akka Cluster Sharding"

Comment (patriknw) [reply docs/src/main/paradox/getting-started/running.md:60]: and the version is wrong for that, not an Akka version:
```
"com.lightbend.akka" %% "akka-cluster-sharding-typed" % "1.0.0-RC1+27-a6e0fa2e"
```

File: examples/src/test/java/jdocs/guide/EventGeneratorApp.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+public class EventGeneratorApp {
+  public static void main(String[] args) throws Exception {
+    Boolean clusterMode = (args.length > 0 && args[0].equals("cluster"));
+    Config config = config();
+    ActorSystem<String> system =
+        ActorSystem.create(Guardian.create(clusterMode), "EventGeneratorApp", config);
+  }
+
+  private static Config config() {
+    return ConfigFactory.parseString("akka.actor.provider = \"cluster\"\n")
```
Comment (patriknw) [inline examples/src/test/java/jdocs/guide/EventGeneratorApp.java:48]: doesn't have to be "cluster", `akka.actor.provider = cluster` is enough

File: docs/src/main/paradox/getting-started/running-cluster.md [hunk 6]
```diff
@@ -20,36 +20,36 @@ Add the `ShoppingCartClusterApp` to your project:
...
 Shopping cart events are tagged in a similar way to the sharded entities themselves.
 Given a sequence of tags from `0..n` a hash is generated using the sharding entity key, the shopping cart id.
 The hash is modded `%` by the number of tags in the sequence to choose a tag from the sequence.
 See the @ref:[Tagging Events in EventSourcedBehavior](../running.md#tagging-events-in-eventsourcedbehavior) section of the documentation for an example of how events can be tagged with Akka Persistence, or look at the source of the `EventGeneratorApp` in the previous @ref:[Running the Projection](running.md) section.
 
-Run the same `EventGeneratorApp` from the previous @ref:[Running the Projection](running.md) section, with an additional argument `cluster`:
+The same `EventGeneratorApp` from the previous @ref:[Running the Projection](running.md) section can be used to generate events for this app with an additional argument `cluster`.
+Run the app with @scala[sbt: `sbt "runMain docs.guide.EventGeneratorApp cluster"`]@java[maven: `mvn compile exec:java -Dexec.mainClass="jdocs.guide.EventGeneratorApp" -Dexec.args="cluster"`].
```
Comment (patriknw) [inline docs/src/main/paradox/getting-started/running-cluster.md:34]: why not tabs for sbt and maven?

File: examples/src/test/java/jdocs/guide/ShoppingCartTags.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+// #guideTags
+package jdocs.guide;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ShoppingCartTags {
+  public static String SINGLE = "shopping-cart";
+  public static List<String> TAGS = Arrays.asList("carts-0", "carts-1", "carts-2");
```
Comment (patriknw) [inline examples/src/test/java/jdocs/guide/ShoppingCartTags.java:13]: or an array?

```
public static String[] TAGS = {"carts-0", "carts-1", "carts-2"};
```

**Review by patriknw — APPROVED** [review]
LGTM after a few small things...

File: docs/src/main/paradox/getting-started/running-cluster.md (hunk 6 above)
Comment (seglo) [reply docs/src/main/paradox/getting-started/running-cluster.md:34]: @patriknw I couldn't figure out how to do tabs for inline snippets.  Can you point me to an example?

Comment (patriknw) [reply docs/src/main/paradox/getting-started/running-cluster.md:34]: Search for `## Dependency` in https://raw.githubusercontent.com/lightbend/akka-enhancements/master/docs/src/main/paradox/persistence-dc/index.md

File: docs/src/main/paradox/getting-started/setup-your-app.md [hunk 8]
```diff
@@ -15,13 +15,38 @@ Add `ShoppingCartEvents` to your project:
 Scala
 :  @@snip [ShoppingCartEvents.scala](/examples/src/test/scala/docs/guide/ShoppingCartEvents.scala) { #guideEvents }
 
+Java
+:  @@snip [ShoppingCartEvents.java](/examples/src/test/java/jdocs/guide/ShoppingCartEvents.java) { #guideEvents }
+
+To enable SerDes of events with Akka Persistence it's necessary to define a base type for your event type hierarchy.
```
[the remark this answers is not in the document]
Comment (seglo) [reply docs/src/main/paradox/getting-started/setup-your-app.md:21]: :+1:

File: docs/src/main/paradox/getting-started/setup-your-app.md [hunk 9]
```diff
@@ -15,13 +15,38 @@ Add `ShoppingCartEvents` to your project:
...
+In this guide we are using [Jackson Serialization](https://doc.akka.io/docs/akka/current/serialization-jackson.html).
+Add the `CborSerializable` base type to your project:
+
+Scala
+:  @@snip [CborSerializable.scala](/examples/src/test/scala/docs/guide/CborSerializable.scala) { #guideCbor }
+
+Java
+:  @@snip [CborSerializable.java](/examples/src/test/java/jdocs/guide/CborSerializable.java) { #guideCbor }
```
[the remark this answers is not in the document]
Comment (seglo) [reply docs/src/main/paradox/getting-started/setup-your-app.md:29]: :+1:

File: docs/src/main/paradox/getting-started/projection-handler.md [hunk 10]
```diff
@@ -54,7 +57,13 @@ Add the following imports to `ShoppingCartApp`:
...
+Java
+:  @@snip [ShoppingCartApp.java](/examples/src/test/java/jdocs/guide/ShoppingCartApp.java) { #guideProjectionImports }
+
 Setup the Projection in the Guardian `Behavior` defined in `ShoppingCartApp`:
 
 Scala
 :  @@snip [ShoppingCartApp.scala](/examples/src/test/scala/docs/guide/ShoppingCartApp.scala) { #guideProjectionSetup }
+
+Java
+:  @@snip [ShoppingCartApp.java](/examples/src/test/java/jdocs/guide/ShoppingCartApp.java) { #guideProjectionSetup }
```
[the remark this answers is not in the document]
Comment (seglo) [reply docs/src/main/paradox/getting-started/projection-handler.md:69]: :+1: Oops, thanks.

File: docs/src/main/paradox/getting-started/running.md (hunk 4 above)
Comment (seglo) [reply docs/src/main/paradox/getting-started/running.md:60]: I see the groupId is wrong for this and probably a few other modules, I'll doublecheck.  Is that what you're referring to?

The version is generated.

File: examples/src/test/java/jdocs/guide/EventGeneratorApp.java (hunk 5 above)
Comment (seglo) [reply examples/src/test/java/jdocs/guide/EventGeneratorApp.java:48]: :+1:

File: examples/src/test/java/jdocs/guide/ShoppingCartTags.java (hunk 7 above)
Comment (seglo) [reply examples/src/test/java/jdocs/guide/ShoppingCartTags.java:13]: :+1:

File: docs/src/main/paradox/getting-started/running.md (hunk 4 above)
Comment (patriknw) [reply docs/src/main/paradox/getting-started/running.md:60]: it's not using the AkkaVersion, which should be 2.5.8

Comment (seglo) [reply docs/src/main/paradox/getting-started/running.md:60]: Got it!  Good catch.

File: docs/src/main/paradox/getting-started/running-cluster.md (hunk 6 above)
Comment (seglo) [reply docs/src/main/paradox/getting-started/running-cluster.md:34]: Perfect

**Pull request merged**